### DISCUSSION TITLE
Symbol definitions on hover

### DISCUSSION THREAD

**Discussion by samuelhward** [JiscSD/octopus#93] [Ideas]
**Problem:**

A common problem in STEM is that of symbols which describe quantities; they are often assumed/sometimes duplicated and can be confusing.

**Proposed solution:**

We could fix this by asking authors to optionally provide definitions for each symbol in their paper. A reader hovering over an equation could then examine the meaning of each symbol in turn. Some information which could be included in the definition could be:

- units 
- derivation
- definition with respect to other quantities
- written description

Instead of users having to write this out for every piece of work, it can be streamlined if we suggest definitions for quantities to users from a database of quantities that have been previously defined by other authors. This database of quantities must be moderated, which might be achieved at scale by making it editable by the community - e.g. wikipedia model.

**Other implications**

If each symbol is its own instance in a piece of work, that could open the door for other features: 

- allowing authors to write equations where each symbol/term/quantity is independently coloured/styled (we could enforce colourblind-friendly colour maps)
- authors could assign these colours quickly to multiple instances of the same symbol which appear in a piece of writing 
- could track the use and evolution of quantities throughout a scientific field
- could search for papers which include a given quantity 
- could search for papers according to their mathematical content, as opposed to their text

**Comment by nasainsbury** [comment]
Hi @samuelhward thanks for the suggestion!

On a very basic level are you thinking that hovering over `π` would display `pi` with a definition?

If so, my questions are:
1. Would there just be a standard list of symbols and definitions that we define/pull from somewhere else?
2. Or would this be a kind of table the user sees, where they can see a list of symbols used, and provide their own definitions?

Apologies if I've completely misunderstood the suggestion.

**Reply by samuelhward** [reply]
Hi @nathan-at-jisc 

No apologies necessary - it's an idea I've left relatively half baked, focusing more on functionality rather than implementation. 

On hover over ω, it would display something like this:

"
Gyro frequency. 
The frequency at which an ion orbits a magnetic field line. 
ω = qB/m. 
[Hz]
"
Which would be written by the user, or selected from some type of drop-down list containing descriptions that other authors have previously defined. So here I pick an example from plasma physics, but equally ω could mean something else in another field, so the user would want to pick or define (if not already) a definition and attach it to that quantity.

As to the format of descriptions, I hadn't thought to standardise this - and I don't think it would be appropriate to enforce it as different fields attach different amounts of information to quantities. E.g. physics quantities usually always have units (kg, metres etc.) whereas in pure maths they often do not. 

@1: It's tricky because it would be good for a user to be able to define their own, but as a speed-up be able to select from a list of pre-defined options that are common in their field that we suggest. That list could be user written/moderated. It's not in anyone's interests to write a poor description as it's going in their own work. 

@2: that could be interesting, although it might take up a lot of visual room - as hover hides the definition information by default

This idea is not a high-priority, but it is just something I wondered _could_ be possible with an OpSci platform such as this. So I'm just registering the idea here for now :)
